datalad/metadata/__init__.py

```diff
@@ -193,7 +193,25 @@ def get_metadata(ds, guess_type=False, ignore_subdatasets=False,
     # XXX it may be worth the put the combined output of this function in a separate
     # cache on the local machine, in order to speed up meta data access, but maybe this
     # is already the domain of a `query` implementation
-    meta.append(get_implicit_metadata(ds, ds_identifier))
+    implicit_meta = get_implicit_metadata(ds, ds_identifier)
+    # create a lookup dict to find parts by subdataset mountpoint
+    has_part = implicit_meta.get('dcterms:hasPart', [])
+    if not isinstance(has_part, list):
+        has_part = [has_part]
+    has_part = {hp['location']: hp for hp in has_part}
+
+    # figure out all other version of this dataset: origin or siblings
+    # build a flat list of UUIDs
+    candidates = ('dcterms:isVersionOf', 'dcterms:hasVersion')
+    ds_versions = [
+        # flatten list
+        item for versionlist in
+        # extract uuids of all listed repos
+        [[implicit_meta[i]['@id'] for r in implicit_meta[i] if '@id' in r]
```

Comment:
Ah, so it is checking if key is @id or @identity or @idiocracy?

On August 12, 2016 3:07:55 PM EDT, Michael Hanke <notifications@github.com> wrote:
>> -    meta.append(get_implicit_metadata(ds, ds_identifier))
>> +    implicit_meta = get_implicit_metadata(ds, ds_identifier)
>> +    # create a lookup dict to find parts by subdataset mountpoint
>> +    has_part = implicit_meta.get('dcterms:hasPart', [])
>> +    if not isinstance(has_part, list):
>> +        has_part = [has_part]
>> +    has_part = {hp['location']: hp for hp in has_part}
>> +
>> +    # figure out all other version of this dataset: origin or
>siblings
>> +    # build a flat list of UUIDs
>> +    candidates = ('dcterms:isVersionOf', 'dcterms:hasVersion')
>> +    ds_versions = [
>> +        # flatten list
>> +        item for versionlist in
>> +        # extract uuids of all listed repos
>> +        [[implicit_meta[i]['@id'] for r in implicit_meta[i] if '@id'
>in r]
>
>r is only the key, not the value, AFAIK
>
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/datalad/datalad/pull/682/files/b8409420b30e186f48d7dd6ea1ab2642b228f969..9bdc084470f9c6f457421613eeb95d942a3ee54e#r74644874